src/software/amazon/ion/impl/IonReaderBinaryRawX.java

```diff
@@ -888,175 +888,210 @@ protected final BigInteger readBigInteger(int len, boolean is_negative) throws I
     }
     protected final int readVarInt() throws IOException
     {
-        int     retvalue = 0;
+        // VarUInt use the last octet as a marker so a 5 byte VarUInt can fit in a Java int.
```

Comment:
refers to VarUInt a couple ltimes, but we're in a VarInt method here.

also, consider starting this comment with:  "VarInt uses the high-order bit of the last octet as a marker;  some (but not all) 5-byte VarInts can fit into a Java int."